### PR TITLE
Generate an embedded token for user

### PR DESCRIPTION
**Pull request by aviadl**
related to https://github.com/descope/etc/issues/2288

### CODE REVIEW

File: descope/api/client.go [hunk 1]
```diff
@@ -231,10 +232,11 @@ type mgmtEndpoints struct {
...
 
-	ssoSettings string
-	ssoMetadata string
-	ssoMapping  string
-	updateJWT   string
+	ssoSettings        string
+	ssoMetadata        string
+	ssoMapping         string
+	updateJWT          string
+	createEmbeddedLink string
```
Comment (itaihanski) [inline descope/api/client.go:239]: I think this should start with `userXXX` and appear at the end of the user functions to keep consistent

File: descope/internal/mgmt/user.go [hunk 2]
```diff
@@ -367,6 +367,29 @@ func (u *user) GenerateEnchantedLinkForTestUser(loginID, URI string) (link, pend
 	return unmarshalGenerateLinkForTestResponse(res)
 }
 
+type embeddedLinkRes struct {
+	Token string `json:"token"`
+}
+
+func (u *user) GenerateEmbeddedLink(loginID string, customClaims map[string]any) (string, error) {
+	if loginID == "" {
+		return "", utils.NewInvalidArgumentError("loginId")
```
Comment (itaihanski) [inline descope/internal/mgmt/user.go:376]:
```suggestion
		return "", utils.NewInvalidArgumentError("loginID")
```

File: descope/sdk/mgmt.go [hunk 3]
```diff
@@ -208,6 +208,10 @@ type User interface {
 	// This is useful when running tests and don't want to use 3rd party messaging services
 	// The redirect URI is optional. If provided however, it will be used instead of any global configuration.
 	GenerateEnchantedLinkForTestUser(loginID, URI string) (link, pendingRef string, err error)
+
+	// Generate an embedded link token, later can be used to authenticate via magiclink verify method
+	// Or via flow verify step
```
Comment (itaihanski) [inline descope/sdk/mgmt.go:213]:
```suggestion
	// or via flow verify step
```

File: README.md [hunk 4]
```diff
@@ -1005,6 +1005,10 @@ link, pendingRef, err := descopeClient.Management.User().GenerateEnchantedLinkFo
 
 // Note 1: The generate code/link methods, work only for test users, will not work for regular users.
 // Note 2: In case of testing sign-in / sign-up methods with test users, need to make sure to generate the code prior calling the sign-in / sign-up methods (such as: descopeClient.Auth.MagicLink().SignUpOrIn)
+
+// Embedded link can be created via the following call,
+// The return value is a token that can be verified via magic link, or using flows
```
Comment (itaihanski) [inline README.md:1010]:
```suggestion
// Embedded links can be created to directly receive a verifiable token without sending it.
// This token that can be verified using the magic link 'verify' function, either directly or through a flow.
```

File: descope/api/client.go (hunk 1 above)
Comment (aviadl) [reply descope/api/client.go:239]: done

File: descope/internal/mgmt/user.go (hunk 2 above)
Comment (aviadl) [reply descope/internal/mgmt/user.go:376]: done

File: descope/sdk/mgmt.go (hunk 3 above)
Comment (aviadl) [reply descope/sdk/mgmt.go:213]: done

File: README.md (hunk 4 above)
Comment (aviadl) [reply README.md:1010]: done

File: README.md [hunk 5]
```diff
@@ -1005,6 +1005,10 @@ link, pendingRef, err := descopeClient.Management.User().GenerateEnchantedLinkFo
 
 // Note 1: The generate code/link methods, work only for test users, will not work for regular users.
 // Note 2: In case of testing sign-in / sign-up methods with test users, need to make sure to generate the code prior calling the sign-in / sign-up methods (such as: descopeClient.Auth.MagicLink().SignUpOrIn)
+
+// Embedded links can be created to directly receive a verifiable token without sending it.
+// This token that can be verified using the magic link 'verify' function, either directly or through a flow.
```
Comment (shilgapira) [inline README.md:1010]:
```suggestion
// This token can then be verified using the magic link 'verify' function, either directly or through a flow.
```

**Review by shilgapira — DISMISSED** [review]

File: README.md (hunk 5 above)
Comment (aviadl) [reply README.md]: Done

**Review by shilgapira — APPROVED** [review]

**Pull request merged**